platform/api/aws/ec2.go

```diff
@@ -200,3 +204,89 @@ func (a *API) CreateTags(resources []string, tags map[string]string) error {
 	}
 	return err
 }
+
+// getSecurityGroupID gets a security group matching the given name.
+// If the security group does not exist, it's created.
+func (a *API) getSecurityGroupID(name string) (string, error) {
+	sgIds, err := a.ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupNames: []*string{&name},
+	})
+	if isSecurityGroupNotExist(err) {
+		return a.createSecurityGroup(name)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("unable to get security group named %v: %v", name, err)
+	}
+	if len(sgIds.SecurityGroups) == 0 {
+		return "", fmt.Errorf("zero security groups matched name %v", name)
```

Comment:
Ok, if it was normal I would have expected it to call create but being an unexpected case an error is fine.